h/schemas/auth_client.py

```diff
@@ -33,7 +34,11 @@ class CreateAuthClientSchema(CSRFSchema):
                                             '"client_credentials" for allowing access to the user creation API'))
 
     trusted = colander.SchemaNode(
-                colander.Boolean(),
```

Comment:
Following some investigation and local testing, I think this should be `colander.Boolean` and so should the `privacy_accepted` field in the other schema. If you make that change you'll also need to change the `privacy_acceptance_validator` to check for `value is False`.

----

The confusion here I think stems from the behavior of `colander.Boolean` perhaps not what you might expect in the sense that any strings (coming from the `POST`-ed form data) _except_ `"false"` or `"0"` are serialized as `False`:

```py
>>> import colander
>>> node = colander.Boolean()
>>> node.deserialize(node, 'false')
False
>>> node.deserialize(node, '0')
False
>>> node.deserialize(node, 'true')
True
>>> node.deserialize(node, '')
True
>>> node.deserialize(node, 'wtf')
True
```

However this works fine with deform's `CheckboxWidget`. When the user submits the form:

1. If the checkbox is *not* checked, then there is no entry for the field in the form data sent to the server. `deform.CheckboxWidget` deserializes this into the string `"false"` and `colander.Boolean` in turn deserializes that to the value `False`.
2. If the checkbox *is* checked, then the form data sent to the server has `trusted=true` which is deserialized by `deform.CheckboxWidget` into the string `"true"` and `colander.Boolean` in turn deserializes that to the value `True`.